### PR TITLE
10837 sentiments list new

### PR DESCRIPTION
**Pull request by propakov**
https://kontur.fibery.io/Tasks/Task/List-with-sentiments-combinations-(Color-legends-section)-dn-fe-10837

Screenshot 2022-07-20 at 16 08 54

### CODE REVIEW

File: src/utils/common/index.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const capitalize = (str) => str.charAt(0).toUpperCase() + str.slice(1);
+
+const toCapitalizedList = (arr: string[]): string =>
+  arr.map(capitalize).join(', ');
+
+const capitalizeArrayOrString = (input: string[] | string): string =>
```
Comment (Akiyamka) [inline src/utils/common/index.ts]: `capitalizeArrayOrString` sound for me like it capitalize that can work with array or string, and it take some time for me for understand that it also join array into one string
In fact this thing doing next
- transform string to array
- capitalize every item in array
- join in string be special way (with space after comma)

Comment (Akiyamka) [inline src/utils/common/index.ts]:
```suggestion
const capitalize = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);

const toCapitalizedList = (arr: string[]): string =>
  arr.map(capitalize).join(', ');

const capitalizeArrayOrString = (input: string[] | string): string =>
```

File: src/utils/common/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  Array.isArray(input) ? toCapitalizedList(input) : capitalize(input);
+
+const sortByKey =
+  <T>(key: string, direction: 'asc' | 'desc') =>
+  (a: T, b: T) =>
+    direction === 'desc' ? b[key] - a[key] : a[key] - b[key];
+
+export { capitalize, toCapitalizedList, capitalizeArrayOrString, sortByKey };
```
Comment (Akiyamka) [inline src/utils/common/index.ts]: should we export `toCapitalizedList`? I'm not found any usage of this

File: src/views/BivariateManager/BivariateManager.module.css [hunk 3]
```diff
@@ -1,5 +1,45 @@
...
+  border-width: 0px 1px 0px 1px;
+  flex-direction: column;
+}
+
+.ListBody {
+  display: flex;
+  flex-grow: 1;
+  overflow: auto;
+  padding-left: 20px;
+  padding-right: 20px;
```
Comment (Akiyamka) [inline src/views/BivariateManager/BivariateManager.module.css]: probably mistake, we usually use 8px grid,

File: src/utils/common/index.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+const toCapitalizedList = (arr: string[]): string =>
+  arr.map(capitalize).join(', ');
+
+const capitalizeArrayOrString = (input: string[] | string): string =>
+  Array.isArray(input) ? toCapitalizedList(input) : capitalize(input);
+
+const sortByKey =
+  <T>(key: string, direction: 'asc' | 'desc') =>
+  (a: T, b: T) =>
+    direction === 'desc' ? b[key] - a[key] : a[key] - b[key];
```
Comment (Akiyamka) [inline src/utils/common/index.ts]:
```suggestion
const sortByKey =
  <T extends Record<string, unknown>>(key: string, direction: 'asc' | 'desc') =>
  (a: T, b: T) =>
    direction === 'desc' ? b[key] - a[key] : a[key] - b[key];
```

File: src/features/bivariate_color_manager/components/SentimentsCombinationsList/SentimentsCombinationsList.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      return {
+        key,
+        maps,
+        verticalLabel,
+        horizontalLabel,
+        legend,
+        id: nanoid(4),
+      };
+    })
+    .sort(sortDescendingByMaps);
```
Comment (Akiyamka) [inline src/features/bivariate_color_manager/components/SentimentsCombinationsList/SentimentsCombinationsList.tsx]: It's expensive code for eval in on every update. I suggests you memoize it or whole component

File: src/features/bivariate_color_manager/components/MiniLegend/MiniLegend.fixture.tsx [hunk 6]
```diff
@@ -1,6 +1,5 @@
-import { useValue } from 'react-cosmos/fixture';
+import React from 'react';
```
Comment (Akiyamka) [inline src/features/bivariate_color_manager/components/MiniLegend/MiniLegend.fixture.tsx]: Are we really need react here?

File: src/utils/common/index.ts (hunk 1 above)
Comment (propakov) [reply src/utils/common/index.ts]: Nice!

File: src/utils/common/index.ts (hunk 2 above)
Comment (propakov) [reply src/utils/common/index.ts]: yeah, now it's not needed

File: src/utils/common/index.ts (hunk 1 above)
Comment (propakov) [reply src/utils/common/index.ts]: True. 
What if 'joinAndCapitalizeItems()' ?

File: src/features/bivariate_color_manager/components/SentimentsCombinationsList/SentimentsCombinationsList.tsx (hunk 5 above)
Comment (propakov) [reply src/features/bivariate_color_manager/components/SentimentsCombinationsList/SentimentsCombinationsList.tsx]: i will be definitely replaced to some atom in future tasks, but yeah, let's do it. 👍

File: src/features/bivariate_color_manager/components/MiniLegend/MiniLegend.fixture.tsx (hunk 6 above)
Comment (propakov) [reply src/features/bivariate_color_manager/components/MiniLegend/MiniLegend.fixture.tsx]: Idk why, but it was required and now it's not, so I've removed it.

**Review by Akiyamka — APPROVED** [review]

**Pull request merged**